src/main/scala/com/normation/plugins/datasources/QueryService.scala

```diff
@@ -177,32 +186,44 @@ class HttpQueryDataSourceService(
                         })
                         //connection timeout: 5s ; getdata timeout: freq ?
         optProperty  <- getHttp.getNode(datasourceId, datasource, nodeInfo, policyServer, parameters, datasource.requestTimeOut, datasource.requestTimeOut)
-                        //on none, don't update anything, the life is wonderful
-        nodeId       <- optProperty match {
-                          case None           => Full(nodeInfo.id)
+        nodeResult   <- optProperty match {
+                          //on none, don't update anything, the life is wonderful (because 'none' means 'don't update')
+                          case None           => Full(NodeUpdateResult.Unchanged(nodeInfo.id))
                           case Some(property) =>
-                            for {
-                              newProps     <- CompareProperties.updateProperties(nodeInfo.properties, Some(Seq(property)))
-                              newNode      =  nodeInfo.node.copy(properties = newProps)
-                              nodeUpdated  <- nodeRepository.updateNode(newNode, cause.modId, cause.actor, cause.reason) ?~! s"Cannot save value for node '${nodeInfo.id.value}' for property '${property.name}'"
-                            } yield {
-                              nodeUpdated.id
+                            // look for the property value in the node to know if an update is needed.
+                            // we only care about value here (not provider or other meta-info)
+                            nodeInfo.properties.find(_.name == property.name).map(_.value) match {
+                              case Some(property.value) => Full(NodeUpdateResult.Unchanged(nodeInfo.id))
```

Comment:
Hard to find that the value of the Node property is equal to value of the datasource property (property.value), maybe easier with changing name of property ('datasourceProperty'?)